src/Microsoft.DotNet.Interactive.HttpRequestParser/Parser/HttpRequestParser.cs

```diff
@@ -356,9 +353,9 @@ private void ConsumeCurrentTokenInto(HttpSyntaxNode node)
                 AddCommentsIfAny(comments, requestNode, addBefore: false);
             }
 
-            if (ParseComment() is { } commentAfterHeaders)
+            foreach (var comment in ParseComments())
             {
-                requestNode.Add(commentAfterHeaders);
+                requestNode.Add(comment);
```

Comment:
Consider: It may be useful to introduce an `HttpSyntaxNode.AddRange` method for comments.